### DISCUSSION TITLE
BeginHorizontal, Spring, Suspend/ResumeLayout not present on docking?

### DISCUSSION THREAD

**Discussion by Emily-TTG** [ocornut/imgui#6458] [New Users Build/Link/Run issues ONLY!]
Heya,

I'm trying to upgrade a codebase which was using a pretty old version of ImGui on the docking branch but it seems like a few features have disappeared? The ones I'm in particular need of are:
```
ImGui::Begin/EndHorizontal
ImGui::Suspend/ResumeLayout
ImGui::Spring

ImGuiContext::NavActivateInputId
ImGuiWindowTempData::CurrentLayout
```

If you could direct me as to how to replicate their behaviours in more up-to-date versions of ImGui?

Thanks!

**Comment by ocornut** [comment]
None of this was ever part of official imgui distribution it is a PR see #846.

I will be working on a way to make this sort of extension entirely external without a mod to main library.

**Comment by philippeflorent** [comment]
great, this is really difficult to try to implement ImGuiNodeEditor based app with such changes

I am trying to use other people api to avoid writing everything from scratch, yet at the end of the day, I wonder if I should not do that :-( pfff

**Reply by ocornut** [reply]
It's not my fault if a third-party extension needs a change to the main library. 
We are still working on integrating necessary logic in main codebase.

However, as I know of https://github.com/thedmd/imgui-node-editor should work with vanilla unmodified imgui.
